tools/flaky-test-retryer/log_parser.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// log_parser.go collects failed tests from jobs that triggered the retryer and
+// finds flaky tests that are relevant to that failed job.
+
+package main
+
+import (
+	"github.com/knative/test-infra/tools/flaky-test-reporter/jsonreport"
+)
+
+// InitLogParser configures jsonreport's dependencies
+func InitLogParser(serviceAccount string) error {
+	return jsonreport.Initialize(serviceAccount)
+}
+
+// getReportRepos gets all of the repositories where we are reporting flaky tests
+// TODO: combine this with the function for getting flaky tests, as they will be
+//       almost exactly the same thing.
+func getReportRepos() ([]string, error) {
+	var repos []string
+	var err error
```

Comment:
this line can be deleted?

Reply:
i need to return err on line 42, otherwise it's out of scope. I could forgo returning an error altogether, if that makes sense?

Reply:
ah. you need to break the combined if then. I like returning errors from lib functions that just silently swallowing them

Reply:
fixed